src/GdbCommand.cc

```diff
@@ -31,11 +31,15 @@ static std::vector<ReplayTimeline::Mark> back_stack;
 static ReplayTimeline::Mark current_history_cp;
 static std::vector<ReplayTimeline::Mark> forward_stack;
 static SimpleGdbCommand rr_history_push(
-    "rr-history-push", [](GdbServer& gdb_server, Task*, const vector<string>&) {
+    "rr-history-push", [](GdbServer& gdb_server, Task* t, const vector<string>&) {
       if (current_history_cp) {
         back_stack.push_back(current_history_cp);
       }
-      current_history_cp = gdb_server.get_timeline().mark();
+      if (!t->session().is_replaying()) {
```

Comment:
isn't this inverted? We want to take a mark if we're replaying, and not if we're not replaying.

But shouldn't we just ignore the command completely if we're not replaying? I don't think you want these other effects.